.github/workflows/tls.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will make use of Faktory put behind NGINX to test 
+# the crate's tls feature (see docker dir in the project's root)
+permissions:
+  contents: read
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+name: tls
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: ubuntu-latest / ${{ matrix.toolchain }} / tls
+    strategy:
+      fail-fast: false
+      matrix:
+        # run on stable and beta to ensure that tests won't break on the next version 
+        # of the rust toolchain
+        toolchain: [stable, beta]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Launch Faktory behind NGINX
+        run: docker compose -f docker/compose.yml up -d
+      - name: Install ${{ matrix.toolchain }}
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.toolchain }}
+      - name: cargo generate-lockfile
+        if: hashFiles('Cargo.lock') == ''
+        run: cargo generate-lockfile
+      - name: Run tests
+        env:
+          FAKTORY_URL: tcp://localhost:7419
+          FAKTORY_URL_SECURE: tcp://localhost:17419
+        run: cargo test --locked --features tls --all-targets
```

Comment:
To speed up CI, I would actually prefer for this job to _only_ run the TLS tests. Which suggests they should live in their own file in `tests/` so that you can pass `--test tls` here.